### PR TITLE
otel-collector: add k8sattributesprocessor

### PR DESCRIPTION
**Pull request by daxmc99**
Part of https://github.com/sourcegraph/sourcegraph/pull/59561#issuecomment-1890116710
## Test plan

CI

### CODE REVIEW

File: docker-images/opentelemetry-collector/builder.template.yaml [hunk 1]
```diff
@@ -36,3 +36,4 @@ processors:
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor v$OTEL_COLLECTOR_VERSION
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor v$OTEL_COLLECTOR_VERSION
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor v$OTEL_COLLECTOR_VERSION
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor v$OTEL_COLLECTOR_VERSION
```
Comment (bobheadxi) [inline docker-images/opentelemetry-collector/builder.template.yaml:39]:
```suggestion
  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor v$OTEL_COLLECTOR_VERSION
```

Note the pattern above, it's not the github URL but the `go.mod` import path

**Comment by daxmc99** [comment]
Ahh thanks @bobheadxi! 
Also super easy to update!

**Comment by bobheadxi** [comment]
FYI, you might want to run `sg ci build docker-images-candidates-notest` over this to make sure the image builds (looks like Bazel doesn't catch it - FYI @sourcegraph/team-dev-infra )

**Review by bobheadxi — APPROVED** [review]

**Comment by daxmc99** [comment]
Building here https://buildkite.com/sourcegraph/sourcegraph/builds/257859

**Pull request merged**